src/psyclone/tests/domain/nemo/transformations/nemo_arrayaccess2loop_trans_test.py

```diff
@@ -601,74 +624,78 @@ def test_validate_iterator():
     the indices of the lhs and rhs arrays do not match, but is OK if they do.
 
     '''
     # Same values
     input_code = (
-        "program test\n"
         "  real :: a(10), b(10)\n"
         "  integer :: n\n"
-        "  a(n+1) = b(1+n)\n"
-        "end program test")
-    reader = FortranReader()
-    psyir = reader.psyir_from_source(input_code)
-    trans = NemoArrayAccess2LoopTrans()
-    index_node = psyir.children[0].children[0].lhs.children[0]
-    trans.apply(index_node)
+        "  a(n+1) = b(1+n)\n")
+    expected_result = (
+        "  real, dimension(10) :: a\n  real, dimension(10) :: b\n"
+        "  integer :: n\n  integer :: ji\n\n"
+        "  do ji = n + 1, n + 1, 1\n"
+        "    a(ji) = b(ji)\n"
+        "  enddo\n\n")
+    check_transformation(tmpdir, input_code, expected_result)
 
     # different values
     input_code = (
-        "program test\n"
         "  real :: a(10), b(10)\n"
         "  integer :: n\n"
-        "  a(n+1) = b(n)\n"
-        "end program test")
-    psyir = reader.psyir_from_source(input_code)
-    trans = NemoArrayAccess2LoopTrans()
-    index_node = psyir.children[0].children[0].lhs.children[0]
+        "  a(n+1) = b(n)\n")
     with pytest.raises(TransformationError) as info:
-        trans.apply(index_node)
+        check_transformation(None, input_code, None)
     assert(
         "Expected index '0' for rhs array 'b' to be the same as the lhs "
         "array 'a', but they differ." in str(info.value))
 
 
-def test_validate_indirection():
-    '''Check that validation works with valid indirection.'''
+def test_validate_indirection(tmpdir):
+    '''Check that validation works with valid indirection and raises an
+    exception with invalid indirection.
 
+    '''
     code = (
-        "program tmp\n"
         "  real :: a(10), b(10)\n"
         "  integer :: lookup(10)\n"
         "  integer :: n\n"
-        "  a(lookup(n)) = b(lookup(n))\n"
-        "end program tmp\n")
-    reader = FortranReader()
-    psyir = reader.psyir_from_source(code)
-    trans = NemoArrayAccess2LoopTrans()
-    index_node = psyir.children[0].children[0].lhs.children[0]
-    trans.validate(index_node)
+        "  a(lookup(n)) = b(lookup(n))\n")
+    expected_result = (
+        "  real, dimension(10) :: a\n  real, dimension(10) :: b\n"
+        "  integer, dimension(10) :: lookup\n  integer :: n\n"
+        "  integer :: ji\n\n"
+        "  do ji = lookup(n), lookup(n), 1\n"
+        "    a(ji) = b(ji)\n"
+        "  enddo\n\n")
+    check_transformation(tmpdir, code, expected_result)
+
+    # different values
+    input_code = (
+        "  real :: a(10), b(10)\n"
+        "  integer :: lookup(10)\n"
+        "  integer :: n\n"
+        "  a(lookup(n)) = b(lookup(1))\n")
+    with pytest.raises(TransformationError) as info:
+        check_transformation(None, input_code, None)
+    assert(
+        "Expected index '0' for rhs array 'b' to be the same as the lhs "
```

Comment:
"...to be the same as that for the lhs..." and please could you include the actual Fortran in the error message?